packages/core/src/model/nodes/ChatNode.ts

```diff
@@ -910,37 +910,25 @@ export class ChatNodeImpl extends NodeImpl<ChatNode> {
             }
 
             if (functionCalls.length > 0) {
-              if (isMultiResponse) {
-                output['function-call' as PortId] = {
-                  type: 'object[]',
-                  value: functionCalls.map((functionCalls) => ({
-                    name: functionCalls[0]?.name,
-                    arguments: functionCalls[0]?.lastParsedArguments,
-                    id: functionCalls[0]?.id,
-                  })),
-                };
-              } else {
-                if (this.data.parallelFunctionCalling) {
-                  console.dir({ functionCalls });
-                  output['function-calls' as PortId] = {
-                    type: 'object[]',
-                    value: functionCalls[0]!.map((functionCall) => ({
-                      name: functionCall.name,
-                      arguments: functionCall.lastParsedArguments,
-                      id: functionCall.id,
-                    })),
-                  };
-                } else {
-                  output['function-call' as PortId] = {
-                    type: 'object',
-                    value: {
-                      name: functionCalls[0]![0]?.name,
-                      arguments: functionCalls[0]![0]?.lastParsedArguments,
-                      id: functionCalls[0]![0]?.id,
-                    } as Record<string, unknown>,
-                  };
-                }
-              }
+							if (isMultiResponse) {
+								output['function-call' as PortId] = {
+									type: 'object[]',
+									value: functionCalls.flat().map((functionCall) => ({
+										name: functionCall.name,
+										arguments: functionCall.lastParsedArguments,
+										id: functionCall.id,
+									})),
+								}
+							} else {
+								output['function-call' as PortId] = {
```

Comment:
The option `parallelFunctionCalling` switches between the `function-call` and `function-calls` output and it looks like that's lost here - this will break existing graphs, unfortunately, could you bring that behavior back?

Reply:
Sorry for the delay, will do